packages/drpc/src/DrpcApi.ts

```diff
@@ -0,0 +1,184 @@
+
+@injectable()
+export class DrpcApi {
+  private drpcMessageService: DrpcService
+  private messageSender: MessageSender
+  private connectionService: ConnectionService
+  private agentContext: AgentContext
+
+  public constructor(
+    messageHandlerRegistry: MessageHandlerRegistry,
+    drpcMessageService: DrpcService,
+    messageSender: MessageSender,
+    connectionService: ConnectionService,
+    agentContext: AgentContext
+  ) {
+    this.drpcMessageService = drpcMessageService
+    this.messageSender = messageSender
+    this.connectionService = connectionService
+    this.agentContext = agentContext
+    this.registerMessageHandlers(messageHandlerRegistry)
+  }
+
+  /**
+   * sends the request object to the connection and returns a function that will resolve to the response
+   * @param connectionId the connection to send the request to
+   * @param request the request object
+   * @returns curried function that waits for the response with an optional timeout in seconds
+   */
+  public async sendRequest(
+    connectionId: string,
+    request: DrpcRequest
+  ): Promise<() => Promise<DrpcResponse | undefined>> {
+    const connection = await this.connectionService.getById(this.agentContext, connectionId)
+    const { requestMessage: drpcMessage, record: drpcMessageRecord } =
+      await this.drpcMessageService.createRequestMessage(this.agentContext, request, connection.id)
+    const messageId = drpcMessage.id
+    await this.sendMessage(connection, drpcMessage, drpcMessageRecord)
+    return async (timeout?: number) => {
+      return await this.recvResponse(messageId, timeout)
+    }
+  }
+
+  /**
+   * Listen for a response that has a thread id matching the provided messageId
+   * @param messageId the id to match the response to
+   * @param timeout the time in seconds to wait for a response
+   * @returns the response object
+   */
+  private async recvResponse(messageId: string, timeout?: number): Promise<DrpcResponse | undefined> {
+    return new Promise((resolve) => {
+      const listener = ({
+        drpcMessageRecord,
+        removeListener,
+      }: {
+        drpcMessageRecord: DrpcRecord
+        removeListener: () => void
+      }) => {
+        const response = drpcMessageRecord.response
+        if (drpcMessageRecord.threadId === messageId) {
+          removeListener()
+          resolve(response)
+        }
+      }
+
+      const cancelListener = this.drpcMessageService.createResponseListener(listener)
+      if (timeout) {
+        const handle = setTimeout(() => {
+          clearTimeout(handle)
+          cancelListener()
+          resolve(undefined)
+        }, timeout * 1000)
+      }
+    })
+  }
+
+  /**
+   * Listen for a request and returns the request object and a function to send the response
+   * @param timeout the time in seconds to wait for a request
+   * @returns the request object and a function to send the response
+   */
+  public async recvRequest(timeout?: number): Promise<
```

Comment:
```suggestion
  public async recvRequest(timeoutMs?: number): Promise<
```

And I promise this is my last comment 😄.

Just for consistency with other timeouts within the framework: it will be good if we can express it in milliseconds

Reply:
no worries lol